File: typescript/collatz-conjecture/collatz-conjecture.ts
class CollatzConjecture {
    static steps(n: number) {
        if (n < 1) throw new Error('Only positive numbers are allowed')
        let count: number = 0
        while (n > 1) {
            if (n % 2 === 0) {
                n /= 2
            } else {
                n = 3 * n + 1
            }
            count++
        }
        return count
    }
}

export default CollatzConjecture

File: dart/collatz-conjecture/lib/collatz_conjecture.dart
class CollatzConjecture {
  int steps(num n) {
    if (n <= 0) throw ArgumentError("Only positive numbers are allowed");
    int count = 0;
    while (n > 1) {
      n = n % 2 == 1 ? 3 * n + 1 : n / 2;
      count++;
    }
    return count;
  }
}
